src/olympia/addons/models.py

```diff
@@ -2072,7 +2069,20 @@ def category(self):
         return CATEGORIES_BY_ID.get(self.category_id)
 
 
+class AddonUserQuerySet(models.QuerySet):
+    def delete(self, hard_delete=False):
+        if hard_delete:
+            return super().delete()
+        else:
+            return self.update(original_role=F('role'), role=amo.AUTHOR_ROLE_DELETED)
+
+    def undelete(self):
+        return self.update(role=models.F('original_role'))
```

Comment:
I thought about it but wasn't sure what - reset back to `AUTHOR_ROLE_DEV` ? I didn't want to introduce a new special value, and I kinda like the idea of keeping a valid value in there in case we accidentally call `undelete()` twice...

Reply:
22737a424fe1323271592cdfaaf605017129a7d2 sets it to the `default` value of the field (currently `AUTHOR_ROLE_DEV`)